src/context/ServiceInfoContext.tsx

```diff
@@ -200,6 +208,60 @@ const ServiceInfoProvider = ({ children }: PropsWithChildren<{}>) => {
     [currentWallet, clearCurrentWallet],
   )
 
+  // Fetch rescan progress when rescanning is true
+  useEffect(() => {
+    if (!serviceInfo?.rescanning || !currentWallet) return
+
+    const abortCtrl = new AbortController()
+    // Flag to track if the API is supported
+    let isRescanProgressApiSupported = true
+
+    const fetchRescanProgress = async (): Promise<void> => {
+      // Skip if API is not supported (returned 404 previously)
+      if (!isRescanProgressApiSupported) return
+
+      try {
+        const res = await Api.getRescanInfo({
+          signal: abortCtrl.signal,
+          token: currentWallet.token,
+          walletFileName: currentWallet.walletFileName,
+        })
+
+        if (res.ok) {
+          const data = await res.json()
+          if (!abortCtrl.signal.aborted && data.progress !== undefined) {
+            dispatchServiceInfo({
+              rescanProgress: data.progress,
+            })
+          }
+        } else if (res.status === 404) {
+          // API not supported (backend version < v0.9.12)
+          isRescanProgressApiSupported = false
+          console.log('Rescan progress API not supported by this backend version')
+        }
+      } catch (err) {
+        if (!abortCtrl.signal.aborted) {
+          console.error('Error fetching rescan progress:', err)
+          // If we get a 404 error, mark the API as not supported
+          if (err instanceof Api.JmApiError && err.response.status === 404) {
+            isRescanProgressApiSupported = false
+            console.log('Rescan progress API not supported by this backend version')
+          }
+        }
+      }
+    }
+
+    fetchRescanProgress()
```

Comment:
We should also initialize with
```
    dispatchServiceInfo({
      rescanProgress: undefined,
    })
```
here?

Othwise it would briefly show the old progress, right? (not tested atm)